src/native/corehost/browserhost/browserhost.cpp

```diff
@@ -0,0 +1,173 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <cstdio>
+#include <vector>
+
+#include <pal.h>
+#include <minipal/utils.h>
+#include <host_runtime_contract.h>
+
+typedef void (*coreclr_error_writer_callback_fn)(const char* line);
+extern "C"
+{
+    pal::hresult_t coreclr_initialize(
+        const char* exePath,
+        const char* appDomainFriendlyName,
+        int propertyCount,
+        const char** propertyKeys,
+        const char** propertyValues,
+        void** hostHandle,
+        unsigned int* domainId);
+
+    pal::hresult_t coreclr_shutdown_2(
+        void* hostHandle,
+        unsigned int domainId,
+        int* latchedExitCode);
+
+    pal::hresult_t coreclr_execute_assembly(
+        void* hostHandle,
+        unsigned int domainId,
+        int argc,
+        const char** argv,
+        const char* managedAssemblyPath,
+        unsigned int* exitCode);
+
+    pal::hresult_t coreclr_create_delegate(
+        void* hostHandle,
+        unsigned int domainId,
+        const char* entryPointAssemblyName,
+        const char* entryPointTypeName,
+        const char* entryPointMethodName,
+        void** delegate);
+
+    pal::hresult_t coreclr_set_error_writer(
+        coreclr_error_writer_callback_fn error_writer);
+
+    const void* SystemResolveDllImport(const char* name);
+    const void* SystemJSResolveDllImport(const char* name);
+    const void* SystemJSInteropResolveDllImport(const char* name);
+    const void* GlobalizationResolveDllImport(const char* name);
+    const void* CompressionResolveDllImport(const char* name);
+
+    bool browserHostExternalAssemblyProbe(const char* pathPtr, /*out*/ void **outDataStartPtr, /*out*/ int64_t* outSize);
```

Comment:
Why are we mixing js-style casing, win32/stdcall-style title casing, and snake_case?

Reply:
I would like to abandon snake_case for JS/TS. But this is where worlds collide.